Demos/include/ApplicationBase.hpp

```diff
@@ -0,0 +1,598 @@
+        _arguments.participantConfiguration = SilKit::Config::ParticipantConfigurationFromString("");
+
+        _commandLineParser = std::make_shared<CommandlineParser>();
+    }
+
+protected:
+    // Must be implemented by the actual application
+
+    // Add application specific command line arguments. Called during SetupCommandLineArgs.
+    virtual void AddCommandLineArgs() = 0;
+    // Evaluate application specific command line arguments. Called during SetupCommandLineArgs.
+    virtual void EvaluateCommandLineArgs() = 0;
+    
+    // Controller creates SIL Kit controllers / services here. 
+    virtual void CreateControllers() = 0;
+
+    // Boot up SIL Kit controllers here.
+    // Called in the CommunicationReadyHandler to ensure that messages emerging from the 
+    // controller creation reach other involved participants.
+    virtual void InitControllers() = 0;
+
+    // When running with the '--async' flag, this is called continuously in a separate worker thread.
+    // This thread is managed by the ApplicationBase. 
+    // The function will trigger as soon as the participant state is 'Running'. 
+    virtual void DoWorkAsync() = 0;
+
+    // When running with time synchronization (no '--async' flag), this is called in the SimulationStepHandler.
+    virtual void DoWorkSync(std::chrono::nanoseconds now) = 0;
+
+private:
+    // Command line parser
+    std::shared_ptr<CommandlineParser> _commandLineParser;
+
+    // Default command line argument identifiers (to allow exclusion)
+    enum DefaultArg
+    {
+        Name,
+        Uri,
+        Log,
+        Config,
+        Async,
+        Duration,
+        AsFastAsPossible
+    };
+    // Names of the default command line arguments.
+    // Note that the 'short name' (e.g. -n) and description are defined at runtime.
+    const std::unordered_map<DefaultArg, std::string> defaultArgName = {
+        {DefaultArg::Name, "name"},
+        {DefaultArg::Uri, "registry-uri"},
+        {DefaultArg::Log, "log"},
+        {DefaultArg::Config, "config"},
+        {DefaultArg::Async, "async"},
+        {DefaultArg::Duration, "duration"},
+        {DefaultArg::AsFastAsPossible, "fast"}};
+
+    // The application arguments. Defaults are set in the constructor.
+    // Values might be overridden by the command line arguments. 
+    struct Arguments
+    {
+        std::string participantName;
+        std::string registryUri;
+        std::shared_ptr<SilKit::Config::IParticipantConfiguration> participantConfiguration;
+        bool runAsync{false};
+        std::chrono::nanoseconds duration;
+        bool asFastAsPossible{false};
+    };
+    Arguments _arguments;
+
+    // SIL Kit API
+    std::unique_ptr<IParticipant> _participant;
+    ILogger* _logger{nullptr};
+    ILifecycleService* _lifecycleService{nullptr};
+    ITimeSyncService* _timeSyncService{nullptr};
+
+    // For sync: wait for sil-kit-system-controller start/abort or manual user abort 
+    enum SystemControllerResult
```

Comment:
See the comment above at `DefaultArg`.

```suggestion
    enum struct SystemControllerResult
```